tiledb/sm/subarray/subarray.cc

```diff
@@ -556,6 +846,89 @@ Status Subarray::get_est_result_size(
   return Status::Ok();
 }
 
+Status Subarray::get_est_result_size_querytype_audited(
```

Comment:
The original Query::get_est_result_size() mostly performed auditing before passing along a request to lower/inner Reader::get_est_result_size() (which calls Subarray::get_est_result_size()) for actual information.
In trying to preserve semantics by bringing that functionality down into Subarray I chose to mnemonically highlight the routine that was really mostly doing auditing before calling another Subarray sibling routine (called just 'get_est_result_size()' as in original query paths) to get the actual information.  The '_querytype' could be removed from the name but is an effort to call attention to the routine's rejection of being used-or-not for certain query types.

Reply:
I think we might want to not expose the size estimations from the subarray since it's already available from the query. This would remove the need for this API altogether.